pkg/controller/resourcequota/resource_quota_controller.go

```diff
@@ -241,8 +241,12 @@ func (rq *ResourceQuotaController) addQuota(obj interface{}) {
 // worker runs a worker thread that just dequeues items, processes them, and marks them done.
 func (rq *ResourceQuotaController) worker(queue workqueue.RateLimitingInterface) func() {
 	workFunc := func() bool {
+		if queue.Len() == 0 {
+			return true
```

Comment:
This change does not make sense. We don't want the workers to perpetually quit and be restarted in steady state… waiting for the next key from the queue is desired.

Reply:
> This change does not make sense. We don't want the workers to perpetually quit and be restarted in steady state… waiting for the next key from the queue is desired.

Okey, but we should look for ways to reduce  ineffective and duplicate update of resource quota. Right?

Reply:
i suppose the problem u r hitting is that:

"frequent quota status updates will fail resource (pod in ur case) creation"

how about linking this topic to our slack channel #sig-api-machinery and see how we could help that and see?

Reply:
> i suppose the problem u r hitting is that:
> 
> "frequent quota status updates will fail resource (pod in ur case) creation"
> 
> how about linking this topic to our slack channel #sig-api-machinery and see how we could help that and see?

we need create and delete a  large number of pods which are under a resource quota. But we always failed to create pod, and the return message is about  `Operation cannot be fulfilled on resourcequotas "quota": the object has been modified; please apply your changes to the latest version and try again`. 
Create and delete pod  will both update resource quota .Create pod will update resource quota in Api-Server, Delete pod  will update resource quota in controller-manager.
When create pods, it maybe will update resource quota failed because of **version conflict** like this `the object has been modified; please apply your changes to the latest version and try again`, 
Api-server will retry 3 times, if failed 3 times ,this pod will fail to create completely and return error.
I want to reduce update resource quota in order to avoid this failure.

Reply:
Did not go through all discussion; but share the same comments with liggt@ as commented at https://github.com/kubernetes/kubernetes/pull/71197#discussion_r238130742